src/operator/nn/layer_norm.cu

```diff
@@ -24,9 +24,662 @@
 */
 #include "./layer_norm-inl.h"
 
+using namespace mshadow::cuda;
+
 namespace mxnet {
 namespace op {
 
+template <typename DType>
+__device__ __forceinline__ DType WARP_SHFL(DType value, int src_lane,
```

Comment:
Why is it capitalized?

Reply:
Because it's warp-level primitives and I feel it's clear to set it as capital-letters. If this is inconsistent, I could change it to be `warp_shfl` and `warp_shfl_xor`.